zproject/computed_settings.py

```diff
@@ -239,6 +239,7 @@ def services_append(service_name: str) -> None:
     "django.middleware.common.CommonMiddleware",
     "zerver.middleware.LocaleMiddleware",
     "zerver.middleware.HostDomainMiddleware",
+    "zerver.middleware.DetectInvalidRequests",
     "zerver.middleware.DetectProxyMisconfiguration",
```

Comment:
I believe this ordering is correct, but this is a very subtle and important part of the change.

Read the comment in DetectProxyMisconfiguration.  Its logic includes the assumption that _Django does not serve content over HTTP_, which we are breaking by making nginx pass `/api/v1` requests to Django.  I think this is _fine_, because by placing the reject-HTTP-api-requests middleware before the DetectProxyMisconfiguration, it is still the case that no HTTP requests will make it to that middleware.

Reply:
Yeah I noticed that while working this so I decided it would be good to place the new middleware before `DetectProxyMisconfiguration`